src/browserlib/extract-headings.js

```diff
@@ -0,0 +1,9 @@
+/**
+ * Extract headings data from documents
+*/
+export default function () {
+  const headings = {};
+  return [...document.querySelectorAll('h1[id], h2[id], h3[id], h4[id], h5[id] ,h6[id]')].map(n => {
+    return {id: n.id, level: n.tagName.slice(1), title: n.textContent.trim().replace(/^[0-9\.]+ /, '').trim()};
```

Comment:
Have you checked whether levels are always meaningful in our specs?

In theory, there could be nested `<section>` that simply reuse `<h1>` throughout. In practice, I suspect editing tools use appropriate headings.

Reply:
I haven't - I'm assuming it will be easier to do once we collect the data :)